File: app/src/main/java/com/popalay/cardme/data/repositories/settings/SettingsRepository.java
package com.popalay.cardme.data.repositories.settings;

import android.content.Context;
import android.support.annotation.NonNull;
import android.util.Log;

import com.github.popalay.rxrealm.RxRealm;
import com.popalay.cardme.data.models.Settings;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Completable;
import io.reactivex.Flowable;
import io.reactivex.Single;

@Singleton
public class SettingsRepository {

    private final Context context;

    @Inject SettingsRepository(Context context) {this.context = context;}

    public Flowable<Settings> listen() {
        return RxRealm.listenElement(realm -> realm.where(Settings.class).findAll());
    }

    public Single<Boolean> hasSettings() {
        return RxRealm.getElement(realm -> realm.where(Settings.class).findFirst())
                .map(settings -> true)
                .toSingle(false);
    }

    public Completable save(@NonNull Settings settings) {
        Log.d("s", "save: " + settings.getLanguage());
        return RxRealm.doTransactional(realm -> realm.copyToRealmOrUpdate(settings));
    }

    public Settings createDefault() {
        return new Settings.Builder()
                .theme("Default")
                .language(context.getResources().getConfiguration().locale.getDisplayLanguage())
                .cardBackground(true)
                .build();
    }
}

File: app/src/main/kotlin/com/popalay/cardme/data/repositories/settings/SettingsRepository.kt
package com.popalay.cardme.data.repositories.settings

import android.content.Context
import com.github.popalay.rxrealm.RxRealm
import com.popalay.cardme.data.models.Settings
import io.reactivex.Completable
import io.reactivex.Flowable
import io.reactivex.Single
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class SettingsRepository @Inject internal constructor(private val context: Context) {

    fun listen(): Flowable<Settings> = RxRealm.listenElement { it.where(Settings::class.java).findAll() }

    fun hasSettings(): Single<Boolean> = RxRealm.getElement { it.where(Settings::class.java).findFirst() }
            .map { true }
            .toSingle(false)

    fun save(settings: Settings): Completable = RxRealm.doTransactional { it.copyToRealmOrUpdate(settings) }

    fun createDefault() = Settings(
            theme = "Default",
            language = "English",
            //language = Locale.getDefault().displayLanguage,
            isCardBackground = true
    )
}
